backuptar/tar.go

```diff
@@ -109,6 +109,72 @@ func BasicInfoHeader(name string, size int64, fileInfo *winio.FileBasicInfo) *ta
 	return hdr
 }
 
+// EncodeSDDLFromTarHeader reads the SDDL associated with the header of the current file
+// from the tar header and encodes it into a byte array.
+func EncodeSDDLFromTarHeader(hdr *tar.Header) ([]byte, error) {
+	// Maintaining old SDDL-based behavior for backward
+	// compatibility.  All new tar headers written by this library
+	// will have raw binary for the security descriptor.
+	var sd []byte
+	var err error
+	if sddl, ok := hdr.PAXRecords[hdrSecurityDescriptor]; ok {
+		sd, err = winio.SddlToSecurityDescriptor(sddl)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if sdraw, ok := hdr.PAXRecords[hdrRawSecurityDescriptor]; ok {
+		sd, err = base64.StdEncoding.DecodeString(sdraw)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return sd, nil
+}
+
+// EncodeExtendedAttributesFromTarHeader reads the EAs associated with the header of the current file
+// from the tar header and encodes it into a byte array.
+func EncodeExtendedAttributesFromTarHeader(hdr *tar.Header) ([]byte, error) {
+	var eas []winio.ExtendedAttribute
+	var eadata []byte
+	var err error
+	for k, v := range hdr.PAXRecords {
+		if !strings.HasPrefix(k, hdrEaPrefix) {
+			continue
+		}
+		data, err := base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			return nil, err
+		}
+		eas = append(eas, winio.ExtendedAttribute{
+			Name:  k[len(hdrEaPrefix):],
+			Value: data,
+		})
+	}
+	if len(eas) != 0 {
+		eadata, err = winio.EncodeExtendedAttributes(eas)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return eadata, nil
+}
+
+// EncodeReparsePointFromTarHeader checks if the given file is a symlink. If yes, then it
+// reads the ReparsePoint structure from the tar header and encodes it into a byte
+// array. If the file is not a symlink then returns an empty byte array.
+func EncodeReparsePointFromTarHeader(hdr *tar.Header) []byte {
```

Comment:
This only cares about (and handles) symlinks so the name is a bit awkward. Would `EncodeSymlinkFromTarHeader` be preferred?

Reply:
I think it is not that this function _only_ handles symlinks. Rather for a file to be a valid reparse point its type should be symlink. So that type check for symlink is more of a verification.
Anyway, I have removed that check from this function since it is already done before calling this function. Let me know what you think about the new definition.

Reply:
My comment could've been worded better, but yes this makes it more apparent what it does and what's needed, thanks! Possibly add in what someone would need to check for on the header to verify they meet this requirement also. e.g. "...must be a symlink (`hdr.Typeflag == tar.TypeSymlink`)"